src/mdx/expandVersionedUrl.js

```diff
@@ -0,0 +1,43 @@
+const tokens = [
+  {
+    token: "$docs$",
+    rules: [
+      // these mappings are currently wrong because there are no optimize versions yet...but eventually, this is what we'll want.
+      { match: "/optimize/", expandTo: "/docs/next" },
+      { match: "/optimize_versioned_docs/version-3.8.0/", expandTo: "/docs" },
+      {
+        match: "/optimize_versioned_docs/version-3.7.0/",
+        expandTo: "/docs/1.3",
+      },
+    ],
+  },
+  {
+    token: "$optimize$",
+    rules: [
+      // these mappings are currently wrong because there are no optimize versions yet...but eventually, this is what we'll want.
+      { match: "/docs/", expandTo: "/optimize/next" },
+      { match: "/versioned_docs/version-8.0/", expandTo: "/optimize" },
+      { match: "/versioned_docs/version-1.3/", expandTo: "/optimize/3.7.0" },
+    ],
+  },
+];
```

Comment:
Thinking ahead here, would we (DX) then add (and adjust) the rules for `8.1` or would we ask the Optimize team to do this? Whose responsibility is it? Feels like DX if we are the ones publishing the release.

Reply:
I would anticipate DX owning this, but that's because I'm assuming we're the ones who cut new versions of the main docs when they're released, too. We'd have to run `docusaurus version` to generate the new version number in addition to changing these rules, and that feels like infrastructure work for _us_. 

Right?

Reply:
Excellent, we are on the same page here. 

🔮 I foresee us updating the docs-on-docs to include clear responsibilities around these minor product releases vs. continuous product releases (when we need to run `docusaurus version` vs. just releasing). These should probably have a clear name too.